### ISSUE TITLE
Confuse with the meaning of combine()

### ISSUE THREAD

**Issue by JodyStats** [ropensci/drake#1065]
I am still experimenting with dynamic branching and I have to say it is a great implementation. Thank you so much for this great feature. One small issue me and my colleagues encountered was the semantics/meaning of `combine()` being used for both splitting and merging operations. Yesterday I showed my plan to one of my colleagues and he actually spent about half an hour just to figure out the `combine()` operations was actually a splitting operation. I also get confused especially when there are many `combine()` used in the plan and mentally it took me a few seconds to scan through the details of the code to figure out whether it is a merging or splitting operation.

My current workaround is to comment out every `combine()` call with "SPLIT ..." and "COMBINE ..." to reduce confusion for both myself and colleagues as shown

```r
drake_plan(
  dataset = gapminder %>% mutate(gdpPercap = scale(gdpPercap)),
  continent = dataset$continent,
  model = target(
    fit_model(dataset),
    dynamic = combine(dataset, .by = continent)   # SPLIT models
  ),
  results = target(
    bind_rows(model),   
    dynamic = combine(model)                      # COMBINE models
  )
)
```

I don't know what is the best solution but I supposed we can use another same function doing the same things with different name? It looks a bit wrong but separating both `combine()` and `split()` (or some better name?) would greatly reduce the confusion? Users have the options to choose whether to stick with `combine()` or use different names.

**Comment by wlandau** [comment]
Yeah, this also came up in #1064. Maybe something does need to be done.

Whether you are splitting a data frame or aggregating the sub-targets of a previous `map()`, the underlying mechanics of dynamic splitting and combining turn out to be identical. So from a development perspective, it makes sense to have only one verb.

But on reflection, the verb does not need to be "combine". Maybe we can find a word that captures both splitting and combining. Some possibilities:

- `group()`
- `regroup()`
- `partition()`
- `chunk()`

`group()` is my favorite so far. It seems to cover both splitting and aggregation, and `target(dynamic = group(.by = ...))` looks kind of like `dplyr::group_by()`.

What do you think? To other early adopters of dynamic branching, feedback on names and other suggestions would be super helpful.

cc @dpmccabe.

**Comment by dpmccabe** [comment]
`group` is the one that makes the most sense to me since it doesn't necessarily imply a subdivision, unlike the others.

**Comment by wlandau** [comment]
Implemented in #1068.
